src/hooks/http.js

```diff
@@ -184,10 +197,17 @@ export const isAlreadyTraced = callback =>
 
 export const httpRequestWrapper = originalRequestFn =>
   function(...args) {
-    const { url, options, callback } = httpRequestArguments(args);
-    const host = getHostFromOptionsOrUrl(options, url);
+    let isTraceDisabled, url, options, callback, host;
+    try {
+      ({ url, options, callback } = httpRequestArguments(args));
+      host = getHostFromOptionsOrUrl(options, url);
+      isTraceDisabled = isBlacklisted(host) || isAlreadyTraced(callback);
+    } catch (err) {
+      logger.warn('request parsing error', err);
+      isTraceDisabled = true;
```

Comment:
Maybe its better the tracer "starting" disabled and you try to change it. if you failed its stay disabled